packages/auth-sveltekit/src/server.ts

```diff
@@ -0,0 +1,489 @@
+import {
+  redirect,
+  type Cookies,
+  type RequestEvent,
+  type Handle,
+} from "@sveltejs/kit";
+import type { Client } from "edgedb";
+import {
+  Auth,
+  builtinOAuthProviderNames,
+  type BuiltinOAuthProviderNames,
+  type TokenData,
+  type emailPasswordProviderName,
+} from "@edgedb/auth-core";
+import {
+  ClientAuth,
+  getConfig,
+  type AuthConfig,
+  type AuthOptions,
+} from "./client.js";
+
+export type { TokenData, AuthOptions, Client };
+
+export type BuiltinProviderNames =
+  | BuiltinOAuthProviderNames
+  | typeof emailPasswordProviderName;
+
+type ParamsOrError<Result extends object, ErrorDetails extends object = {}> =
+  | ({ error: null } & { [Key in keyof ErrorDetails]?: undefined } & Result)
+  | ({ error: Error } & ErrorDetails & { [Key in keyof Result]?: undefined });
+
+export interface AuthRouteHandlers {
+  onOAuthCallback(
+    params: ParamsOrError<{
+      tokenData: TokenData;
+      provider: BuiltinOAuthProviderNames;
+      isSignUp: boolean;
+    }>
+  ): Promise<Response>;
```

Comment:
```suggestion
  ): Promise<never>;
```
I think these handlers all should throw a redirect(), so the return type should be `never`.